src/js/structs/Framework.js

```diff
@@ -42,7 +43,11 @@ module.exports = class Framework extends Service {
     let url = this.get('webui_url');
 
     if (url != null && url !== '') {
```

Comment:
Could we do, this in stead of the nested if?
```js
getWebURL() {
  let url = this.get('webui_url');
  let name = super.getName();
  
  if (!url || !name) {
    return null;
  }
  
  return `${Config.rootUrl}/service/${name}`;
}
```

Also, I used `!url` in stead of `url != null && url !== ''` since it will handle all those cases (and `false` or `0` won't make sense either)